server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -176,7 +176,15 @@ protected ExecutorService createThreadPoolExecutor() {
         masterService.setClusterStatePublisher((clusterStatePublicationEvent, publishListener, ackListener) -> {
             clusterStateRef.set(clusterStatePublicationEvent.getNewState());
             ClusterServiceUtils.setAllElapsedMillis(clusterStatePublicationEvent);
-            publishListener.onResponse(null);
+            threadPool.executor(randomFrom(ThreadPool.Names.SAME, ThreadPool.Names.GENERIC)).execute(() -> {
+                final var threadName = Thread.currentThread().getName();
+                try {
+                    Thread.currentThread().setName("[" + MasterService.MASTER_UPDATE_THREAD_NAME + "]");
```

Comment:
This seems unnecessary now that we use the `ThreadedActionListener`. I think we should remove this line (and the try-finally), that would validate that we can call the publish listener from any thread.

Reply:
Thanks, yes - see 464358cb7360de13b64cbf48f0f646069540be4a.